File: src/core/menu/menues/MainMenu.java
package core.menu.menues;

import core.Main;
import core.menu.Menu;
import core.menu.MenuComponent;
import core.menu.components.*;
import misc.Debug;
import misc.math.*;

public class MainMenu extends Menu
{
	public MainMenu()
	{
		getComponents().add(new Label(this, new Rect(400, 10, 200, 40), "Main Menu"));
		getComponents().add(new EditField(this, new Rect(200, 200, 100, 100), "text"));
		getComponents().add(new Button(this, new Rect(300, 100, 100, 30), "Create Server")
		{
			@Override public void onClick(int mouseButton)
			{
				Main.getMenues().add(new CreateServerMenu());
			}
		});
		getComponents().add(new Button(this, new Rect(300, 150, 100, 30), "Join Server")
		{
			@Override public void onClick(int mouseButton)
			{
				Main.getMenues().add(new JoinServerMenu());
			}
		});
		getComponents().add(new Button(this, new Rect(140, 500, 100, 30), "Logout")
		{
			@Override public void onClick(int mouseButton)
			{
				Main.getMenues().remove(Main.getMenues().getLast());
			}
		});
		getComponents().add(new QuitButton(this, new Rect(20, 500, 100, 30)));
	}

}

File: src/menu/menues/MainMenu.cpp
#include "MainMenu.h"

#include "../../core/Main.h"
#include "../components/EditField.h"
#include "../components/Button.h"
#include "../components/Label.h"
#include "CreateServerMenu.h"
#include "JoinServerMenu.h"

MainMenu::MainMenu()
{
	addComponent(new Label(this, PixelRect(400, 10, 200, 40), "Main Menu"));
	addComponent(new EditField(this, PixelRect(200, 200, 100, 100), "text"));

	class CreateServerButton : public Button
	{
		public:
			CreateServerButton(MainMenu* c, const PixelRect& r, const std::string& s) : Button(c, r, s) {}
			void onClick(int mouseButton)
			{
				Main::getMenuList()->addMenu(new CreateServerMenu());
			}
	};
	addComponent(new CreateServerButton(this, PixelRect(300, 100, 100, 30), "Create Server"));

	class JoinServerButton : public Button
	{
		public:
			JoinServerButton(MainMenu* c, const PixelRect& r, const std::string& s) : Button(c, r, s) {}
			void onClick(int mouseButton)
			{
				Main::getMenuList()->addMenu(new JoinServerMenu());
			}
		
	};
	addComponent(new JoinServerButton(this, PixelRect(300, 150, 100, 30), "Join Server"));

	class LogoutButton : public Button
	{
		public:
			LogoutButton(MainMenu* c, const PixelRect& r, const std::string& s) : Button(c, r, s) {}
			void onClick(int mouseButton)
			{
				Main::logout();
				Main::getMenuList()->back();
			}
	};
	addComponent(new LogoutButton(this, PixelRect(140, 500, 100, 30), "Logout"));

	class QuitButton : public Button
	{
		public:
			QuitButton(MainMenu* c, const PixelRect& r, const std::string& s) : Button(c, r, s) {}
			void onClick(int mouseButton)
			{
				Main::exit();
			}
	};
	addComponent(new QuitButton(this, PixelRect(20, 500, 100, 30), "Quit"));
}
